internal/services/storage/storage_account_network_rules_resource_test.go

```diff
@@ -151,6 +151,34 @@ func TestAccStorageAccountNetworkRules_empty(t *testing.T) {
 	})
 }
 
+func TestAccStorageAccountNetworkRules_redeploy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_account_network_rules", "test")
+	r := StorageAccountNetworkRulesResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.deploy(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That("azurerm_storage_account.test").ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.remove(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That("azurerm_storage_account.test").ExistsInAzure(r),
+			),
+		},
+		{
```

Comment:
can we add an import step here?

Reply:
Hi @katbyte , this step(**r.remove(data)**) will delete the **azurerm_storage_account_network_rules** resource. If we add **ImportStep()** here, the system will report error of **resource id not exist**.

Reply:
Ah, in this case can we add a custom test step that checks to ensure the resource has actually been correctly deleted? thanks!

Reply:
we'll want to confirm the Exists function does the same here too

Reply:
Hi @katbyte , @tombuildsstuff , I have updated the test cases and all of them passed(Except **TestAccStorageAccountNetworkRules_SynapseAccess**, which is already failed in main branch). Kindly help review it again. Thanks.